ddpui/ddpairbyte/airbytehelpers.py

```diff
@@ -243,6 +213,76 @@ def create_connection(org: Org, payload: AirbyteConnectionCreate):
         orgtask=org_task,
     )
 
+    return org_dataflow
+
+
+def create_connection(org: Org, payload: AirbyteConnectionCreate):
+    """creates an airbyte connection and tracking objects in the database"""
+    warehouse = OrgWarehouse.objects.filter(org=org).first()
+    if warehouse is None:
+        return None, "need to set up a warehouse first"
+    if warehouse.airbyte_destination_id is None:
+        return None, "warehouse has no airbyte_destination_id"
+    payload.destinationId = warehouse.airbyte_destination_id
+
+    if warehouse.airbyte_norm_op_id is None and not warehouse.is_destinations_v2():
+        warehouse.airbyte_norm_op_id = airbyte_service.create_normalization_operation(
+            org.airbyte_workspace_id
+        )["operationId"]
+        warehouse.save()
+
+    org_airbyte_server_block = OrgPrefectBlockv1.objects.filter(
+        org=org,
+        block_type=AIRBYTESERVER,
+    ).first()
+    if org_airbyte_server_block is None:
+        raise Exception(f"{org.slug} has no {AIRBYTESERVER} block in OrgPrefectBlock")
+
+    sync_task = Task.objects.filter(slug=TASK_AIRBYTESYNC).first()
+    if sync_task is None:
+        return None, "sync task not supported"
+
+    reset_task = Task.objects.filter(slug=TASK_AIRBYTERESET).first()
+    if reset_task is None:
+        return None, "reset task not supported"
+
+    airbyte_conn = airbyte_service.create_connection(
+        org.airbyte_workspace_id, warehouse.airbyte_norm_op_id, payload
+    )
+
+    try:
+        with transaction.atomic():
+
+            # create a sync deployment & dataflow
+            org_task = OrgTask.objects.create(
+                org=org, task=sync_task, connection_id=airbyte_conn["connectionId"]
+            )
+
+            sync_dataflow: OrgDataFlowv1 = create_airbyte_deployment(
+                org, org_task, org_airbyte_server_block
+            )
+
+            # create reset connection deplpyment & dataflow
+            org_task = OrgTask.objects.create(
+                org=org, task=reset_task, connection_id=airbyte_conn["connectionId"]
+            )
+
+            reset_dataflow: OrgDataFlowv1 = create_airbyte_deployment(
+                org, org_task, org_airbyte_server_block
+            )
+
+            sync_dataflow.reset_conn_dataflow = reset_dataflow
+            sync_dataflow.save()
+
+    except Exception as err:
+        # delete the airbyte connection; since the deployment didn't get created
+        logger.info("deleting airbyte connection")
+        airbyte_service.delete_connection(
+            org.airbyte_workspace_id, airbyte_conn["connectionId"]
+        )
```

Comment:
shouldn't we clean up any OrgTask / OrgDataFlowv1 which may have been created? or does the transaction.atomic automatically roll them back